File: Iniciante/1094 - Experiências/Python/1094.py
def main():
    n = int(input())
    
    total_guinea_pigs = 0
    total_rabbits = 0
    total_rats = 0
    total_frogs = 0

    for _ in range(n):
        number, letter = input().split()
        number = int(number)

        total_guinea_pigs += number

        if letter == 'C':
            total_rabbits += number
        elif letter == 'R':
            total_rats += number
        elif letter == 'S':
            total_frogs += number

    percentage_rabbits = total_rabbits / total_guinea_pigs * 100
    percentage_rats = total_rats / total_guinea_pigs * 100
    percentage_frogs = total_frogs / total_guinea_pigs * 100

    print(f"Total: {total_guinea_pigs} cobaias")
    print(f"Total de coelhos: {total_rabbits}")
    print(f"Total de ratos: {total_rats}")
    print(f"Total de sapos: {total_frogs}")
    print(f"Percentual de coelhos: {percentage_rabbits:.2f} %")
    print(f"Percentual de ratos: {percentage_rats:.2f} %")
    print(f"Percentual de sapos: {percentage_frogs:.2f} %")

if __name__ == "__main__":
    main()

File: Iniciante/1094 - Experiências/TypeScript/1094.ts
import * as fs from "fs";

function main(): void {
    const input: string[] = fs.readFileSync(0, 'utf8').trim().split('\n');
    const n: number = parseInt(input[0], 10);

    let totalGuineaPigs: number = 0;
    let totalRabbits: number = 0;
    let totalRats: number = 0;
    let totalFrogs: number = 0;

    for (let i = 1; i <= n; i++) {
        const [numberStr, letter] = input[i].trim().split(/\s+/);
        const number: number = parseInt(numberStr, 10);

        totalGuineaPigs += number;

        switch (letter) {
            case 'C':
                totalRabbits += number;
                break;
            case 'R':
                totalRats += number;
                break;
            case 'S':
                totalFrogs += number;
                break;
        }
    }

    const percentageRabbits: number = totalGuineaPigs > 0 ? (totalRabbits / totalGuineaPigs) * 100 : 0;
    const percentageRats: number = totalGuineaPigs > 0 ? (totalRats / totalGuineaPigs) * 100 : 0;
    const percentageFrogs: number = totalGuineaPigs > 0 ? (totalFrogs / totalGuineaPigs) * 100 : 0;

    console.log(`Total: ${totalGuineaPigs} cobaias`);
    console.log(`Total de coelhos: ${totalRabbits}`);
    console.log(`Total de ratos: ${totalRats}`);
    console.log(`Total de sapos: ${totalFrogs}`);
    console.log(`Percentual de coelhos: ${percentageRabbits.toFixed(2)} %`);
    console.log(`Percentual de ratos: ${percentageRats.toFixed(2)} %`);
    console.log(`Percentual de sapos: ${percentageFrogs.toFixed(2)} %`);
}

main();